### ISSUE TITLE
Focus ring behavior within modal

### ISSUE THREAD

**Issue by bedrich-schindler** [react-ui-org/react-ui#584]
Before re-implementation Modal from `<div>` to `<dialog>`, focus ring was always visible. After the change, focus ring is visible on the next focused item. This has something to do with autofocus functionality natively provided by HtmlDialogElement, see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog#accessibility

With @mbohal, we are not sure whether it is correct behavior or not. I think that it behaves natively and we do not have to change it. It would probably require to change focus-ring mixin that uses `:focus-visible` instead of `:focus`.

cc @adamkudrna

**Comment by adamkudrna** [comment]
Do you mean the focus ring is not there when you open the dialog with mouse, but when it's open and you hit the tab, you can see it? If so, it's alright. It depends on _how_ you open the dialog:

- with mouse click ➡ focus ring is not visible until tab is hit,
- with keyboard ➡ focus ring is visible right after the dialog is being opened.

Browser decides the best if the focus ring should be visible or not. We just use the `:focus-visible` pseudo-selector for that in CSS.

**Comment by lukasbriza** [comment]
After a call with @mbohal, it was decided that the current Modal implementation has the correct behavior we want. This includes the focus trap as well as the mentioned focus ring. The correct behavior aligns with what @adamkudrna described.
